test/test_memparser.py

```diff
@@ -0,0 +1,207 @@
+        sig = Signal()
+        # The starting 's:' is removed by eclipseMemstructGen.py parsing
+        raw_line = f'{name}, {datatype}, {initvalue}, {factor}, {offset}, {unit}, {minvalue}, {maxvalue}'
+        get_signal_from_entry(raw_line, index, sigid, sig)
+        index += 1
+
+        # Test that parsing was successful
+        assert sig.name == name
+        assert sig.id == sigid
+        assert sig.type == types_dict[datatype]
+        assert sig.bitsize == bitsizes_dict[datatype]
+        assert sig.float == ('true' if datatype == 'F' else 'false')
+        assert sig.signed == signed_dict[datatype]
+        assert sig.init_value == initvalue
+        assert sig.factor == factor
+        assert sig.offset == offset
+        assert sig.unit == unit
+        assert sig.minvalue == minvalue
+        assert sig.maxvalue == maxvalue
+
+
+def test_get_message_from_entry():
+    # Test message parsing
+    # Message format :
+    # m:<Message Name>, <ID>
+
+    # To add a test case, simply add a value to the tables :
+    names_to_test = ['NAME']
+    ids_to_test = [0x01, 0x10]
+    values_to_test = itertools.product(names_to_test, ids_to_test)
+
+    index = 0
+    for name, message_id in values_to_test:
+        message = Message()
+        # The starting 'm:' has already been parsed by eclipseMemstructGen.py
+        raw_line = '{}, {:#02x}'.format(name, message_id)
+        get_message_from_entry(raw_line, index, message)
+        index += 1
+
+        # Test that the parsing was successful
+        assert message.name == name
+        assert message.id == message_id
+
+
+def test_get_board_from_entry():
+    # Test board parsing
+    # Board format :
+    # 'b:<Board Name>, <Offset>, <Extend>, <Endianness>'
+    # Offset = ID, Extend = Bool for extended CAN, Endianness = 1 for little endian, 0 for big endian
+
+    names_to_test = ['NAME']
+    offsets_to_test = [0x0, 0x100]
+    extends_to_test = [0, 1]  # 0 = standard CAN, 1 = extended CAN
+    endianness_to_test = [0, 1]  # 0 = Big endian, 1 = little endian
+    values_to_test = itertools.product(names_to_test, offsets_to_test, extends_to_test, endianness_to_test)
+
+    index = 0
+    for name, offset, extend, endianness in values_to_test:
+        board = Board()
+        # The starting 'b:' has already been parsed by eclipseMemstructGen.py
+        offset_string = '{:#02x}'.format(offset)  # Formatting for hex ('0xAA')
+        raw_line = f'{name}, {offset_string}, {extend}, {endianness}'
+        get_board_from_entry(raw_line, index, board)
+        index += 1
+
+        # Test that parsing was successful
+        assert board.name == name
+        assert board.offset == offset
+        assert board.extend == extend
+        assert board.little_endian == endianness
+
+
+def check_signal_report_error(capsys, name, datatype, init_value, factor, offset, unit_name, min_value, max_value):
```

Comment:
```suggestion
def assert_signal_report_error(capsys, name, datatype, init_value, factor, offset, unit_name, min_value, max_value):
```

Le mot `check` pourrait laisser à croire que cette fonction est dans le vrai code et qu'elle fait de la validation. Avec `assert` on est certain qu'il s'agit d'un assert comme les autres.